org.milkytracker.MilkyTracker.yaml

```diff
@@ -0,0 +1,65 @@
+app-id: org.milkytracker.MilkyTracker
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+command: milkytracker
+finish-args:
+  - --device=dri
+  - --share=ipc
+  - --socket=pulseaudio
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --filesystem=xdg-music
+
+modules:
+  - name: rtmidi
+    buildsystem: cmake-ninja
+    sources:
+      - type: archive
+        url: https://github.com/thestk/rtmidi/archive/refs/tags/6.0.0.tar.gz
+        sha256: ef7bcda27fee6936b651c29ebe9544c74959d0b1583b716ce80a1c6fea7617f0
+        x-checker-data:
+          type: anitya
+          project-id: 4220
+          url-template: https://github.com/thestk/rtmidi/archive/refs/tags/$version.tar.gz
+
+  - name: lhasa
+    sources:
+      - type: archive
+        url: https://github.com/fragglet/lhasa/releases/download/v0.4.0/lhasa-0.4.0.tar.gz
+        sha256: a7fc883c304c508562fb93fa307a4c342b0c886fcc265f28b92dc0c39220c5b3
+        x-checker-data:
+          type: anitya
+          project-id: 14822
+          url-template: https://github.com/fragglet/lhasa/releases/download/v$version/lhasa-$version.tar.gz
+
+  - name: zziplib
+    buildsystem: cmake-ninja
+    sources:
+      - type: archive
+        url: https://github.com/gdraheim/zziplib/archive/refs/tags/v0.13.72.tar.gz
+        sha256: 93ef44bf1f1ea24fc66080426a469df82fa631d13ca3b2e4abaeab89538518dc
+        x-checker-data:
+          type: anitya
+          project-id: 13802
+          url-template: https://github.com/gdraheim/zziplib/archive/refs/tags/v$version.tar.gz
+
+  - name: milkytracker
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+    sources:
+      - type: git
+        url: https://github.com/milkytracker/MilkyTracker.git
+        tag: v1.04.00
+        commit: 7e9171488fc47ad2de646a4536794fda21e7303d
+        x-checker-data:
+          type: git
+          tag-pattern: ^v([\d.]+)$
+      - type: patch
+        path: milkytracker.appdata.patch
+    post-install:
+      - install -Dm644 resources/pictures/carton.png ${FLATPAK_DEST}/share/icons/hicolor/128x128/apps/org.milkytracker.MilkyTracker.png
```

Comment:
you should also add the install rules into your upstrea submissions. There are supposed to be installed by the build system. It's a bug if they are not.

Reply:
by these, include .desktop and appstream file.

Reply:
I agree, but I am afraid that this goes beyond my CMake skills and also as far as I know, the CMake file is used not only on Linux/*nix, but on various non-Unix systems including for example [MorphOS](https://www.morphos-team.net). I don't want to possibly break things. What I will however do is creating a bug report ticket for this.

Reply:
like virtually 100% of CMake using project that build on Linux.

BTW if I stopped at things i don't know I'd never be here.

Reply:
https://github.com/milkytracker/MilkyTracker/issues/336

I may try writing the rules myself later in upcoming days - weeks, but i don't promise anything since my free time is very limited. For now, I am glad that I could make the Flatpak manifest (and the other two ones that I recently made) and that everything works fine.

Reply:
@kowalski7cc Do you want to try working on this? :-)